tests/forms_tests/widget_tests/test_selectdatewidget.py

```diff
@@ -319,7 +319,7 @@ def test_selectdate_required_placeholder(self):
             field = DateField(widget=SelectDateWidget(years=('2018', '2019')), required=required)
             self.check_html(field.widget, 'my_date', '', html=(
                 """
-                <select name="my_date_month" id="id_my_date_month" %(m_placeholder)s>
+                <select name="my_date_month" id="id_my_date_month">
```

Comment:
For the `required=True` case, should there not be a `required` attribute here, which would then trigger the HTML validation error?